addons/editor/initKeybinds.sqf

```diff
@@ -113,6 +113,17 @@
     };
 }, {}, [DIK_R, [true, true, false]]] call CBA_fnc_addKeybind; // Default: CTRL + SHIFT + R
 
+[ELSTRING(main,DisplayName), QGVAR(pingCurators), [LSTRING(pingCurators), LSTRING(pingCurators_Description)], {
+    if (!isNull curatorCamera && {!GETMVAR(RscDisplayCurator_search,false)}) then {
+        private _thisCurator = getAssignedCuratorLogic player;
+        private _object = (SELECTED_OBJECTS + [_thisCurator]) select 0;
+        if (_object == _thisCurator) then {_thisCurator setPosASL ([] call EFUNC(common,getPosFromScreen))};
```

Comment:
How about creating a game logic?

Reply:
I think that would be worse. I am not sure what BIS_fnc_curatorPinged` does but maybe we can just expand on its functionality to support pinging objects and positions.

Reply:
```sqf
/*
	Author: Karel Moricky

	Description:
	Show effects when player pings curator.
	Icon and entity list highlighting is handled by engine.

	Parameter(s):
		0: OBJECT - curator module
		1: OBJECT - player

	Returns:
	BOOL
*/

//--- Terminate when not enough thime passed since the last ping
private ["_time"];
_time = missionnamespace getvariable ["bis_fnc_curatorPinged_time",0];
if (time - _time < 0.1) exitwith {};
missionnamespace setvariable ["bis_fnc_curatorPinged_time",time];

//--- Get arguments
private ["_curator","_player","_pitch","_source","_soundPing","_sound","_volume","_pitchBase"];
_curator = _this param [0,objnull,[objnull]];
_player = _this param [1,objnull,[objnull]];
_isCurator = getassignedcuratorunit _curator == player;

//--- Mark position of the last ping (for external use)
missionnamespace setvariable ["bis_fnc_curatorPinged_player",_player];

//--- Play pinging sound
_soundsPing = getarray (configfile >> "cfgCurator" >> "soundsPing");
if (count _soundsPing > 0 || !_isCurator) then {
	_sound = "";
	if (_isCurator) then {
		//--- Curator - assign unique sound to each player
		_n = 0;
		{_n = _n + _x;} foreach (toarray name _player);
		_sound = _soundsPing select (_n % (count _soundsPing))
	} else {
		//--- Store that player already pinged, so hint is not displayed anymore
		if !(isnil {missionnamespace getvariable "bis_fnc_curatorPinged_index"}) then {
			profilenamespace setvariable ["bis_fnc_curatorPinged_done",true];
			saveprofilenamespace;
		};

		//--- Player - pick random sound to create a melody (ever played Joruney? :)
		_soundsPingCount = count _soundsPing;
		_index = missionnamespace getvariable ["bis_fnc_curatorPinged_index",floor random _soundsPingCount];
		_index = round ((_index - 2 + random 4) + _soundsPingCount) % _soundsPingCount;
		_sound = _soundsPing select _index;
		missionnamespace setvariable ["bis_fnc_curatorPinged_index",_index];
	};
	playsound [_sound,true];
};

//--- Show animated icon in the middle of the screen for player
if !(_isCurator) then {
	([] call bis_fnc_rscLayer) cutrsc ["RscCuratorPing","plain"];
};

true
```

Reply:
then used here
```sqf
// P:\a3\ui_f_curator\UI\Displays\RscDisplayCurator.sqf
case (_key in actionkeys 'curatorPingView'): { //--- Go to last ping
    _target = missionnamespace getvariable "bis_fnc_curatorPinged_player";
    if !(isnil "_target") then {
        _pos = getposatl _target;
        _posZ = (getposatl curatorcamera select 2) max 5;
        _pos set [2,(_pos select 2) + _posZ];
        _pos = [_pos,_posZ * 2,direction curatorcamera + 180] call bis_fnc_relpos;
        [_pos,_target] spawn bis_fnc_setcuratorcamera;
    };
    _return = true;
};
```

Reply:
> I dislike the side effect of moving the curator logic object. I guess if we want to avoid that, we would need to add our own ping system.

I mean, if not this, what else could the location of the curator logic object be used for?

Reply:
> > I dislike the side effect of moving the curator logic object. I guess if we want to avoid that, we would need to add our own ping system.
> 
> I mean, if not this, what else could the location of the curator logic object be used for?

Currently, ALiVE mod causes problems as that updates the curator logic object's position every second. Not our fault but I think it is best to avoid this requirement.

I added an alternative approach that also improves the visuals by making the ping grow and fade out.